hazelcast/src/main/java/com/hazelcast/spi/tenantcontrol/TenantControl.java

```diff
@@ -65,5 +64,29 @@
      * Hazelcast object from the tenant
      * This is so the TenantControl itself can be garbage collected
      */
-    void unregister();
+    void distributedObjectDestroyed();
+
+    /**
+     * Checks if tenant app is loaded at the current time and classes are available
+     *
+     * @param op passed so the tenant can filter on who is calling
+     * @return true if tenant is loaded and classes are available
+     */
+    boolean isAvailable(Operation op);
+
+    /**
+     * clean up the thread to avoid potential class loader leaks
+     */
+    void clearThreadContext();
```

Comment:
What's the difference between this method and close? I would expect close to clear the context.

Reply:
added more comments. This is catch-all method that clears all of the potential context instead of just the one that was used in setTenant()